### PR TITLE
feat: add rpc endpoints `eth_getBlockByNumber` & `eth_getBlockByHash`

### LINKED ISSUE #31: Implement eth_getBlockByHash and eth_getBlockByNumber RPC endpoints
[eth_getBlockByHash - RPC Spec](https://ethereum.github.io/execution-apis/api-documentation/#eth_getBlockByHash)

[eth_getBlockByNumber - RPC Spec](https://ethereum.github.io/execution-apis/api-documentation/#eth_getBlockByNumber)

### PR DESCRIPTION
**Pull request by fmoletta**
Based on #150, please merge it first

**Motivation**

Comply with rpc spec for `eth_getBlockByNumber` & `eth_getBlockByHash`

**Description**
* Integrate `Store` as a state for the rpc api

* Add RPC endpoints `eth_getBlockByNumber` & `eth_getBlockByHash`

Closes #31

### CODE REVIEW

File: crates/rpc/src/lib.rs [hunk 1]
```diff
@@ -71,13 +88,20 @@ pub fn map_requests(req: &RpcRequest) -> Result<Value, RpcErr> {
...
-pub async fn handle_http_request(body: String) -> Json<Value> {
+pub async fn handle_http_request(State(storage): State<Store>, body: String) -> Json<Value> {
     let req: RpcRequest = serde_json::from_str(&body).unwrap();
 
     let res: Result<Value, RpcErr> = match req.method.as_str() {
         "eth_chainId" => client::chain_id(),
         "eth_syncing" => client::syncing(),
-        "eth_getBlockByNumber" => block::get_block_by_number(),
+        "eth_getBlockByNumber" => {
+            let request = GetBlockByNumberRequest::parse(&req.params).unwrap();
```
Comment (mpaulucci) [inline crates/rpc/src/lib.rs:98]: can we return an RpcError here instead of unwrapping?

Comment (fmoletta) [reply crates/rpc/src/lib.rs:98]: We need to refactor `handle_http_request` to work like `handle_authrpc_request` in order to propagate the error, Ill make a small PR for it

Comment (fmoletta) [reply crates/rpc/src/lib.rs:98]: Upate: Here it is: #156

**Review by MegaRedHand — APPROVED** [review]
Great work!!

**Pull request merged**